### PR TITLE
Removing platform mri 2.3

### PR DESCRIPTION
**Pull request by srijitamukherjee**
**What this PR does / why we need it**:

> Removing mri_23 - mri_25

**Which issue(s) this PR fixes** 

We do not have nay open issue for this.

**Special notes for your reviewer**:

Adding WIP as in case, there are more ruby 2.3 related stuff which needs to be removed, we can do it on the same PR.

### CODE REVIEW

File: Gemfile.base [hunk 1]
```diff
@@ -91,10 +91,9 @@ gem 'analytics-ruby', require: false
 group :development, :test do
   gem 'bootsnap', '~> 1.3'
 
-  platform :mri_23, :mri_24, :mri_25 do
-    gem 'pry-byebug', '>= 3.7.0', require: false
-    gem 'pry-stack_explorer', require: false
-  end
+  gem 'pry-byebug', '>= 3.7.0', require: false
```
Comment (hallelujah) [inline Gemfile.base]: You can keep ruby 2.4 and 2.5 because it does not work on JRuby

Comment (srijitamukherjee) [reply Gemfile.base]: Okay, I will add it back

File: Gemfile.base [hunk 2]
```diff
@@ -91,10 +91,10 @@ gem 'analytics-ruby', require: false
 group :development, :test do
   gem 'bootsnap', '~> 1.3'
 
-  platform :mri_23, :mri_24, :mri_25 do
+  platform :mri_24, :mri_25 do
     gem 'pry-byebug', '>= 3.7.0', require: false
     gem 'pry-stack_explorer', require: false
-  end
+end
```
Comment (duduribeiro) [inline Gemfile.base]: Just align this `end` with the `platform`

Comment (Martouta) [inline Gemfile.base]:
```suggestion
  end
```

File: Gemfile.base [hunk 3]
```diff
@@ -91,10 +91,10 @@ gem 'analytics-ruby', require: false
 group :development, :test do
   gem 'bootsnap', '~> 1.3'
 
-  platform :mri_23, :mri_24, :mri_25 do
+  platform :mri_24, :mri_25 do
```
Comment (Martouta) [inline Gemfile.base]: Now we don't support 2.4 anymore so it should be:
```suggestion
  platform :mri_25 do
```

File: Gemfile.base (hunk 2 above)
Comment (srijitamukherjee) [reply Gemfile.base]: Fixed

**Pull request merged**